patterns/Creational/template/Kotlin/template.kt

```diff
@@ -0,0 +1,47 @@
+// Abstract class defining the template method and steps
+abstract class NamesTemplate {
+    // Template method to manage names
+    fun manageNames(name: String) {
+        println("Starting to manage the name: $name")
+        addName(name)
+        checkName(name)
+        listNames()
+        println("Finished managing the name: $name\n")
+    }
```

Comment:
In your example, when adding a name there are a lot of events messages that get triggered after you call `manageNames`. Although I don't think this is necessarily the wrong approach, I feel like it could hinder the experience, as it prints every action that takes place.

I think there should be a better way of doing it, where you only get one success message saying "$name has been successfully added to the database".

Because as of right now a user will get this message per name added to the database.
`Starting to manage the name: $name`,
`Added: $name to the database.`
`Checked: $name exists in the database` or `Checked: $name does not exist in the database`
`Current names in the database: [Array]`
`Finished managing the name: $name`